src/vm/stackwalk.cpp

```diff
@@ -1778,7 +1778,7 @@ StackWalkAction StackFrameIterator::Filter(void)
                                 pCurrTracker = pCurrTracker->GetPreviousExceptionTracker();
                             }
 
-                            if (pCurrTracker != NULL)
+                            if ((pCurrTracker != NULL) && !pCurrTracker->IsEHClauseFilter())
```

Comment:
If this is not the correct fix, then it should be changed to an assert to detect this unexpected condition.